src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1560,12 +1560,15 @@ LedgerMaster::shouldAcquire (
     std::uint32_t const ledgerHistoryIndex,
     std::uint32_t const candidateLedger) const
 {
-    // If the ledger could be the current ledger acquire it.
-    // Otherwise, acquire it only if it's within our range of
-    // desired history and we wouldn't delete it if we had it.
-    bool ret = (candidateLedger >= currentLedger) ||
-        ((candidateLedger >= ledgerHistoryIndex) &&
-        ((currentLedger - candidateLedger) <= ledgerHistory));
+
+    // Fetch ledger if it might be the current ledger,
+    // is requested by the advisory delete setting, or
+    // is within our configured history range
+
+    bool ret (candidateLedger >= currentLedger ||
+        ((ledgerHistoryIndex > 0) &&
+            (candidateLedger > ledgerHistoryIndex)) ||
+        (currentLedger - candidateLedger) <= ledgerHistory);
```

Comment:
However, I _do_ think this one should be `(currentLedger - candidateLedger) < ledgerHistory`.
Example: ledger history is 100, meaning we only want 100 ledgers. The current ledger is 1100, so we want ledgers 1001-1100. candidate ledger 1000 would give `1100 - 1000 = 100`, and `100 <= ledgerHistory` is true, but it should be false.